src/en/authors-charm-hooks.md

```diff
@@ -19,37 +19,23 @@ the unit agent's log. If it returns a non-zero exit code, the agent enters an
 The agent will also enter an error state if the unit agent process is aborted
 during hook execution.
 
-There are two types of hooks, described in more detail in the following
-sections.
+There are multiple types of hooks, each described in more detail in the
+following sections.
 
-!!! Note: None of the unit or relation hooks are required; if you don't
-implement a hook, it just doesn't get run. When a hook event occurs, Juju will
-look for the corresponding hook file to execute, but if it finds none, will
-continue running without generating an error.
+!!! Note: None of the hooks are required; if you don't implement a hook, it
+just doesn't get run. When a hook event occurs, Juju will look for the
+corresponding hook file to execute, but if it finds none, will continue
+running without generating an error.
 
-## Unit hooks
+All the hooks must be written to be
```

Comment:
@chuckbutler What do you think of me making the reference to idempotency more universal to hooks, rather than keeping it as a brief mention in the install hook section? After your comment, I got to thinking and this seems to make sense to me. Hopefully it is also less pedantic.